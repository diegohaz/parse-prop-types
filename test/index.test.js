import PropTypes from 'prop-types'
import parsePropTypes from '../src'

const propTypes = {
  foo: PropTypes.string.isRequired,
  bar: PropTypes.bool,
  baz: PropTypes.arrayOf(PropTypes.string).isRequired,
  qux: PropTypes.oneOf(['foo', 'bar', 'baz']),
  quux: PropTypes.shape({
    foo: PropTypes.string,
    bar: PropTypes.instanceOf(String).isRequired,
  }),
  foo2: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({
      foo: PropTypes.string,
    }),
  ]),
}

describe('parsePropTypes', () => {
  it('works', () => {
    const Component = {
      propTypes,
      defaultProps: {
        bar: false,
      },
    }
    expect(parsePropTypes(Component)).toMatchSnapshot()
  })

  it('works without defaultProps', () => {
    const Component = { propTypes }
    expect(parsePropTypes(Component)).toMatchSnapshot()
  })

  it('works without propTypes', () => {
    const Component = {}
    expect(parsePropTypes(Component)).toEqual({})
  })
})
