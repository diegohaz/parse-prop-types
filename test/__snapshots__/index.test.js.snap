// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parsePropTypes works 1`] = `
Object {
  "bar": Object {
    "defaultValue": Object {
      "value": false,
    },
    "required": false,
    "type": Object {
      "name": "bool",
    },
  },
  "baz": Object {
    "required": true,
    "type": Object {
      "name": "arrayOf",
      "value": Object {
        "name": "string",
      },
    },
  },
  "foo": Object {
    "required": true,
    "type": Object {
      "name": "string",
    },
  },
  "foo2": Object {
    "required": false,
    "type": Object {
      "name": "oneOfType",
      "value": Array [
        Object {
          "name": "string",
        },
        Object {
          "name": "shape",
          "value": Object {
            "foo": Object {
              "required": false,
              "type": Object {
                "name": "string",
              },
            },
          },
        },
      ],
    },
  },
  "quux": Object {
    "required": false,
    "type": Object {
      "name": "shape",
      "value": Object {
        "bar": Object {
          "required": true,
          "type": Object {
            "name": "instanceOf",
            "value": "String",
          },
        },
        "foo": Object {
          "required": false,
          "type": Object {
            "name": "string",
          },
        },
      },
    },
  },
  "qux": Object {
    "required": false,
    "type": Object {
      "name": "oneOf",
      "value": Array [
        "foo",
        "bar",
        "baz",
      ],
    },
  },
}
`;

exports[`parsePropTypes works without defaultProps 1`] = `
Object {
  "bar": Object {
    "required": false,
    "type": Object {
      "name": "bool",
    },
  },
  "baz": Object {
    "required": true,
    "type": Object {
      "name": "arrayOf",
      "value": Object {
        "name": "string",
      },
    },
  },
  "foo": Object {
    "required": true,
    "type": Object {
      "name": "string",
    },
  },
  "foo2": Object {
    "required": false,
    "type": Object {
      "name": "oneOfType",
      "value": Array [
        Object {
          "name": "string",
        },
        Object {
          "name": "shape",
          "value": Object {
            "foo": Object {
              "required": false,
              "type": Object {
                "name": "string",
              },
            },
          },
        },
      ],
    },
  },
  "quux": Object {
    "required": false,
    "type": Object {
      "name": "shape",
      "value": Object {
        "bar": Object {
          "required": true,
          "type": Object {
            "name": "instanceOf",
            "value": "String",
          },
        },
        "foo": Object {
          "required": false,
          "type": Object {
            "name": "string",
          },
        },
      },
    },
  },
  "qux": Object {
    "required": false,
    "type": Object {
      "name": "oneOf",
      "value": Array [
        "foo",
        "bar",
        "baz",
      ],
    },
  },
}
`;
